### PR TITLE
Fix runtime error from split barrier

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Before this PR, when enabling split barrier, `RuntimeError: Triton Error [ZE]: 0x78000018`.

This PR changes device library module metadata from
```
!spirv.Source = !{!1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1}
!1 = !{i32 4, i32 100000}
```
to
```
!spirv.Source = !{!1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1}
!1 = !{i32 3, i32 100000}
```
Note:
`4` in `!spirv.Source = !{!1} !1 = !{i32 4, i32 100000}` corresponds to OpenCL C++, 
so OpenCL C builtins e.g., `_Z29intel_work_group_barrier_waitii` is not translated to SPIRV builtins, 
which causes IGC to not recognize the split barrier builtins and not lowered them to GenISA.
`3` in `!spirv.Source = !{!1} !1 = !{i32 3, i32 100000}` corresponds to OpenCL C.

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
Will measure the performance impact before merging in the PR.

File: third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc
Comment (victor-eds) [inline third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc:1]: What are the changes here?

Comment (whitneywhtsang) [reply third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc:1]: As mentioned in the PR description, 
Change device library module metadata from
```
!spirv.Source = !{!1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1}
!1 = !{i32 4, i32 100000}
```
to
```
!spirv.Source = !{!1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1, !1}
!1 = !{i32 3, i32 100000}
```

Comment (whitneywhtsang) [reply third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc:1]: so just changed a character.

**Review by victor-eds — APPROVED** [review]

File: third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc
Comment (victor-eds) [reply third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.bc:1]: Excellent!

**Comment by whitneywhtsang** [comment]
> Will measure the performance impact before merging the PR.

Enabling split barrier doesn't give performance improvement in geomean for different problem sizes of GEMM, 
so we are not going to enable split barrier in this PR.

**Pull request merged**